### ISSUE TITLE
GetType has the same problem as Bridge.

### ISSUE THREAD

**Issue by michaelcheers** [sq/JSIL#981]
See [#1301](https://github.com/bridgedotnet/Bridge/issues/1301) in Bridge.
[Dot Net Fiddle](https://dotnetfiddle.net/SS4SJr)
This code gives an incorrect result:
```csharp
using System;
/// <summary>
/// Creates a simple class
/// </summary>
public class App
{
    // Use the [Ready] attribute to automaically
    // run this method when the page is ready.
    //[Ready]
    public static void Main()
    {
        object v = (byte)3;
        Console.WriteLine(v.GetType().FullName);
        v = (uint)3;
        Console.WriteLine(v.GetType().FullName);
        v = (ushort)3;
        Console.WriteLine(v.GetType().FullName);
        v = (short)3;
        Console.WriteLine(v.GetType().FullName);
    }
}
```

**Comment by iskiselev** [comment]
icsharpcode/ILSpy#708 is root cause of it.

**Comment by iskiselev** [comment]
Resolved via #982.
